components/broker/engine/engine.go

```diff
@@ -480,9 +473,15 @@ func (brk *broker) findTarget(ctx *BrokerEventContext) (err error) {
 				if ctx.TargetAdapter, err = brk.store.Adapter(ctx, ctx.Event.Target.Name, typ); err != nil {
 					return err
 				}
-				if err := ctx.TargetAdapter.Resolve(ctx.Data); err != nil {
+				// TODO: Data on `ctx.Event.Target` still need to be filled out, however,
```

Comment:
@xadhatter I am struggling to figure out exactly how we want to accomplish the goal of removing the hacky code in `sub_mgr.go#Subscription`

In order to find a group subscription I need get past:
`if comp == nil || comp.Name == "" || comp.Hash == "" {`

This means that in order to match to a group subscription we need what the `Hash` of the component is at this point in time. I couldn't see a readily apparent way to doing this lookup at this point in time.

Should we remove the Hash check that is required in order to match to a group subscription. This code:

```
func (mgr *subscriptionMgr) GroupSubscription(comp *core.Component) (GroupSubscription, bool) {
	if comp == nil || comp.Name == "" {
		return nil, false
	}

	mgr.mutex.RLock()
	defer mgr.mutex.RUnlock()

	grp, found := mgr.grpMap[comp.GroupKey()]
	if !found || grp == nil || !grp.IsActive() {
		return nil, false
	}

	return grp, true
}
```

Reply:
Actually - Removing the Hash from the group subscription check wouldn't work without more refactoring. Since GroupKey assumes that the Hash exists:
```
func (c *Component) GroupKey() string {
	if c == nil {
		return ""
	}
	return utils.Join("-", c.App, c.Name, c.ShortHash())
}
```